lib/puppet/indirector/json.rb

```diff
@@ -52,7 +52,7 @@ def load_json_from_file(file, key)
     json = nil
 
     begin
-      json = File.read(file)
+      json = File.read(file).force_encoding(Encoding::UTF_8)
```

Comment:
All FileSystem calls should be going through https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_system/file_impl.rb#L74